src/geometry/euclidean/angle/unoriented/affine.lean

```diff
@@ -439,4 +439,53 @@ lemma angle_lt_pi_of_not_collinear {p₁ p₂ p₃ : P} (h : ¬collinear ℝ ({p
   ∠ p₁ p₂ p₃ < π :=
 (angle_le_pi _ _ _).lt_of_ne $ angle_ne_pi_of_not_collinear h
 
+/-- The sine of the angle between three points is zero if and only if the angle is 0 or π. -/
+lemma sin_eq_zero_iff_angle_eq_zero_or_angle_eq_pi {p₁ p₂ p₃ : P} :
+  real.sin (∠ p₁ p₂ p₃) = 0 ↔ ∠ p₁ p₂ p₃ = 0 ∨ ∠ p₁ p₂ p₃ = π :=
+begin
+  refine ⟨λ h, _, λ h, _⟩,
+  swap, cases h; rw h; simp,
+  rw real.sin_eq_zero_iff at h,
+  cases h with n h,
+  have h₀ : (0 : ℝ) ≤ ↑n :=
+    le_of_mul_le_mul_right
+      (calc 0 * π = 0            : zero_mul π
+              ... ≤ (∠ p₁ p₂ p₃) : angle_nonneg _ _ _
+              ... = n * π        : h.symm)
+      real.pi_pos,
+  have h₁ : (↑n : ℝ) ≤ 1 :=
+    le_of_mul_le_mul_right
+      (calc ↑n * π = (∠ p₁ p₂ p₃) : h
+                ... ≤ π            : angle_le_pi _ _ _
+                ... = 1 * π        : (one_mul π).symm)
+      real.pi_pos,
+  norm_cast at h₀ h₁,
+  rw [le_iff_lt_or_eq, int.lt_iff_add_one_le, zero_add] at h₀,
+  cases h₀,
+  rw antisymm h₁ h₀ at h, right, simpa using h.symm,
+  rw  ← h₀          at h, left,  simpa using h.symm,
+end
```

Comment:
I think this proof is a lot longer than necessary. I'd suggest starting by using `lt_or_eq` on `angle_le_pi`, to split into cases where the angle is equal to or less than π. If equal to π you're done, and if less than π you can use sin_eq_zero_iff_of_lt_of_lt (since that the angle is greater than -π follows from `angle_nonneg`). So you don't need to use the more general `sin_eq_zero_iff` which is responsible for most of the complexity in this proof.

Reply:
I added the corresponding cosine lemmas as well, and in place of this long proof, used `sin_eq_zero_iff_cos_eq`